atomic_reactor/utils/operator.py

```diff
@@ -87,153 +192,138 @@ def replace_pullspecs(self, replacement_pullspecs):
 
         :param replacement_pullspecs: mapping of pullspec -> replacement
         """
-        self._replace_related_image_pullspecs(replacement_pullspecs)
-        self._replace_env_pullspecs(replacement_pullspecs)
-        self._replace_container_image_pullspecs(replacement_pullspecs)
-        self._replace_annotation_pullspecs(replacement_pullspecs)
-        self._replace_init_container_pullspecs(replacement_pullspecs)
+        named_pullspecs = self._named_pullspecs()
+
+        for p in named_pullspecs:
+            old = ImageName.parse(p.image)
+            new = replacement_pullspecs.get(old)
+
+            if new is not None and old != new:
+                log.debug("%s - Replaced pullspec for %s: %s -> %s",
+                          self.path, p.description, old, new)
+                p.image = new.to_str()  # `new` is an ImageName
 
     def replace_pullspecs_everywhere(self, replacement_pullspecs):
         """
         Replace all pullspecs found anywhere in data
 
         :param replacement_pullspecs: mapping of pullspec -> replacment
         """
-        log_template = "%(path)s - Replaced pullspec: %(old)s -> %(new)s"
         for k in self.data:
-            self._replace_pullspecs_everywhere(self.data, k, replacement_pullspecs, log_template)
+            self._replace_pullspecs_everywhere(self.data, k, replacement_pullspecs)
 
-    def _get_related_images(self):
+    def set_related_images(self):
+        """
+        Find pullspecs in predefined locations and put all of them in the
+        .spec.relatedImages section (if it already exists, clear it first)
+        """
+        named_pullspecs = self._named_pullspecs()
+
+        by_name = OrderedDict()
+        conflicts = []
+
+        for new in named_pullspecs:
+            # Keep track only of the first instance with a given name.
+            # Ideally, existing relatedImages should come first in the list,
```

Comment:
There should be no existing relatedImages, you can assert that if you want :)

Reply:
I don't think `OperatorCSV` should worry about this, it makes sense to me to re-include existing relatedImages if there are any. IMO whether we use it like that or not should be up to the plugin

Reply:
ok